src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -575,25 +563,31 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
          * Otherwise, we do nothing. */
         if (!has_xbootldr && slash_boot_exists()) {
                 r = slash_boot_in_fstab();
+                if (r < 0)
+                        return r;
+                if (r > 0)
+                        return 0;
```

Comment:
Why?? A dedicated non-XBOOTLDR partition is absolutely valid. And we should still mount the ESP to /efi/ if possible nonetheless.

Reply:
@YHNdnzj I am not sure we should. I don't like the mix&match here tbh. It sounds to me people should use either the automatic logic for boot partitions or the manual logic. But trying to somehow find a way to mix the two concepts sounds like something that can only fail.

The two dirs are not orthogonal after all (unlike the other mounts), given that we might mount either to /boot under certain conditions.